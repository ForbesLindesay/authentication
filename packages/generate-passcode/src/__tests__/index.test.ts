import generatePassCode, {Encoding} from '../';

jest.mock('crypto', () => ({
  randomBytes(count: number, cb: (err: null, buf: Buffer) => void) {
    const seed = Buffer.from(
      'OxhR9el5IAHiwyAuaoWb9I/8+pVCu9zE0uE8sR779wOjDagq+U3lIZX3n2lmqa97S9qSmUnSHCwIuAc/R5xX0wnjG+UT8vCpApqYserXQGMhRZ73lxLv/7Imw4v3gqsHPwOW9YGOZtNf7RsIz/xct3UPILJyiLyhCTIK2CSFhItpycdya3f9kGZ7gDOTbJOmXKIrarFYqE0e4e3iqvqw34nbxvXEmlPQpSf/wqCKENZspIg1xfXcQQoapKBDmHWLKgDRcDXHWelXu0XeQ3gDMBz434HOsnqbh5e9Y32BUMIkZTtXEKihRNxU+0Wpmo4DcojyyJDJdmxffdKTEAiSiLHeEa5OeRgUfvLvRrlDiEQnhC59dAWYmuKKayuxatikA3OMH8IsW0KqeE7YoFHGJaXEl94LfQ8UlCgX/syTdvYTdDRThkhsMZ3dgnkmhyvoFgw74difZGEZmrMY+Cmye1mEu+FVEwEDkd9IZmV9GPqP2kST5cxC1yUdzYz/pPz2a4oOO+RUls4RjWp7ZzCgfBX7yKcpIlOt+a6x3SbM1oFTyzBV5UhinSfHYQsfQbKytL/djFfwBdvZanvJpsIp+uT1Wnkhj/3hrCZsW86DHC6dql2PbwZrB+1dXz//aUsD0hyBcp4AWypSuepv9tsaw6u+accEnvKm/JQyu4NnOSh3CKB2VEOUOuT2Tntg/rOM5RSFTh7jkJ2zpFdLZYb+n4Jwm3RISKlT8/j0jP0+FkbVSKAJYGxIBDZ0GqZgDPplnOCXcPRpWL/i5rt3gUnspQVLU5ezywTSKgyEvQEha/fVDg0pN2tA4jIw/TJRHFdQFIra8W0KGxQeRW5ZX2Pwb0ledFeH91Uz6z5ZbGyYpMBokdA3jT4BJuQIA15LM1BBtx5/Vc++6P5zjv47Uf8EXVdRmiffdyOFFKez7OAZzD8kxZKVL9iY2vqPEe5xjO01nomONaoK4r1EQ+5GVYH/XGD4pzS7heLD5PfJwbmNYprW9y+nOCKNzYb+ebbr4voEDNiu+TImyHNq//K/W4M5omnNsK66w0AnUA44qYE3wzSLUFHet0oAIyeABn2wmojcN2z5xZAbeeGOz6GOmaopLIxvamlvxC0X/VCOKbcWs0eKQyQXWxCzrLJJwJtPcSDhEPlhzr4STdXR7YVcK/u2lTOvpQ7odSPy2Z+VrnzOV/L0CLVbK+1QNl3CZwwuPjFadbY1kxL6o7IoxR/3yb1BR/NzukR8JzXyTQoxmmYKHEs+IQiQRqUYR+tbM4KTXYPrPlAFOtuz4MPt2Lw9JnnjYMQ7FD00DNAuOC55RrIvxWOSwFPWjEc7IJsl',
      'base64',
    );
    const result = Buffer.allocUnsafe(count);
    seed.copy(result, 0, 0, count);
    return cb(null, result);
  },
}));

test('generatePassCode(40)', async () => {
  const passCode = await generatePassCode(40);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, Encoding.decimal)', async () => {
  const passCode = await generatePassCode(40, Encoding.decimal);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, Encoding.hex)', async () => {
  const passCode = await generatePassCode(40, Encoding.hex);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, Encoding.base32)', async () => {
  const passCode = await generatePassCode(40, Encoding.base32);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, Encoding.base64)', async () => {
  const passCode = await generatePassCode(40, Encoding.base64);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, Encoding.base91)', async () => {
  const passCode = await generatePassCode(40, Encoding.base91);
  expect(passCode.length).toBe(40);
  expect(passCode).toMatchSnapshot();
});

test('generatePassCode(40, {custom})', async () => {
  expect(
    await generatePassCode(40, {
      upperCaseLetters: true,
      lowerCaseLetters: true,
      numbers: true,
      symbols: true,
    }),
  ).toMatchSnapshot('everything');
  expect(
    await generatePassCode(40, {
      upperCaseLetters: true,
      lowerCaseLetters: true,
      numbers: true,
      symbols: false,
    }),
  ).toMatchSnapshot('no symbols');
  expect(
    await generatePassCode(40, {
      upperCaseLetters: true,
      lowerCaseLetters: true,
      numbers: false,
      symbols: true,
    }),
  ).toMatchSnapshot('no numbers');
  expect(
    await generatePassCode(40, {
      upperCaseLetters: true,
      lowerCaseLetters: false,
      numbers: true,
      symbols: true,
    }),
  ).toMatchSnapshot('no lower case letters');
  expect(
    await generatePassCode(40, {
      upperCaseLetters: false,
      lowerCaseLetters: true,
      numbers: true,
      symbols: true,
    }),
  ).toMatchSnapshot('no upper case letters');
});
