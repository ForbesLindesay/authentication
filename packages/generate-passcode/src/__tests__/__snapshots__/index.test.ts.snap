// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`everything 1`] = `"P)5dv:9CU\\"(vs?9#$mv~/};9h/T%0P$nX*JjS|rI"`;

exports[`generatePassCode(40) 1`] = `"OxhR9el5IAHiwyAuaoWb9I8pVCu9zE0uE8sR779w"`;

exports[`generatePassCode(40, Encoding.base32) 1`] = `"HMMFD5PJPEQADYWDEAXGVBM36SH7Z6UVIK55ZRGS"`;

exports[`generatePassCode(40, Encoding.base64) 1`] = `"OxhR9el5IAHiwyAuaoWb9I8pVCu9zE0uE8sR779w"`;

exports[`generatePassCode(40, Encoding.base91) 1`] = `"P)5dv:9CU\\"(vs?9#$mv~/};9h/T%0P$nX*JjS|rI"`;

exports[`generatePassCode(40, Encoding.decimal) 1`] = `"3185159792001232026859489542421311703308"`;

exports[`generatePassCode(40, Encoding.hex) 1`] = `"3b1851f5e9792001e2c3202e6a859bf48ffcfa95"`;

exports[`no lower case letters 1`] = `"P)5:9CU\\"(?9#$~/};9/T%0P$X*JS|ISBA4>\`QP)5"`;

exports[`no numbers 1`] = `"P)dv:CU\\"(vs?#$mv~/};h/T%P$nX*JjS|rISBAe>"`;

exports[`no symbols 1`] = `"P5dv9CUvs9mv9hT0PnXJjSrISBAe4dQP5dv9CUvs"`;

exports[`no upper case letters 1`] = `")5dv:9\\"(vs?9#$mv~/};9h/%0$n*j|re4>d\`)5dv"`;
